backends/dpdk/dpdkHelpers.cpp

```diff
@@ -134,27 +135,58 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
                 src1Op = src1Member;
             }
         }
+        /* If a constant is present in binary operation and it is not 8-bit
+           aligned and it is signed, then the constant value is sign extended
+           with a width of either 32 bit if its original bit-width is less than
+           32 bit or 64 bit if the bit-width of constant is greater than 32 bit
+        */
+
+        if (!isEightBitAligned(src2Op) && src2Op->is<IR::Constant>()) {
```

Comment:
Shouldn't sign-extension be done for non-constant values too? Otherwise I think that overflow handling will be incorrect.

Reply:
Added shift operation for sign extension for non-constant value